src/mscorlib/shared/System/Diagnostics/Tracing/EventSourceException.cs

```diff
@@ -18,9 +18,6 @@ namespace System.Diagnostics.Tracing
     /// <summary>
     /// Exception that is thrown when an error occurs during EventSource operation.
     /// </summary>
-#if !ES_BUILD_PCL
-    [Serializable]
```

Comment:
I believe that the EventSourceException is Serializable because we require all Exceptions to be serializable on the Desktop runtime to support Exceptions traveling across AppDomain boundaries.     

EventSource is unusual, in that at least at one time we tried to keep the source code the same on Desktop and CoreCLR (however because shiproom would wanted only changes that were in response to a particular request to change the desktop, the code bases are NOT in sync today).    Still it may still be useful to keep open the option of a 'bulk update' (I think it is still true that we COULD simply copy the contents to the desktop and it would be 'correct'.   Removing [Serializable] for the desktop build would break that. 

I looked and we don't have a #ifdef that means' Desktop.    The closest thing we have is 'not coreCLR or Project N or not Portable' 

#if  !CORECLR && !ES_BUILD_PN && !ES_BUILD_PCL

This is my recommendation.   Simply surround the [serialable] with this instead.